### ISSUE TITLE
Custom post-receive hook fails with exec format error

### ISSUE THREAD

**Issue by wlmost** [gogits/gogs#4774]
- Gogs version (or commit ref): 0.11.4
- Git version:  git 1.8.3.1
- Operating system:  CentOS Linux release 7.2.1511
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X ] Not relevant
- Log gist (usually found in `log/gogs.log`):

## Description
I use git hook for running jenkins builds after submitting commits. After upgrade to gogs 0.11.4 
I got following error
`{ t } master » git push origin master
Counting objects: 9, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (7/7), done.
Writing objects: 100% (9/9), 926 bytes | 926.00 KiB/s, done.
Total 9 (delta 5), reused 0 (delta 0)
remote: Gogs: Internal error
remote: Fail to execute custom post-receive hook: fork/exec /services/gogs-repositories/xxxxxxxx/sapbti_artmas.git/custom_hooks/post-receive: exec format error
To http://xxx.xxx.xxx.xxx:3000/xxxxxxxx/sapbti_artmas.git
   d4c945b..7a1af27  master -> master
`

I'm working on a other users repository.

Following an excerpt of gogs.log at this time
`
2017/09/15 15:47:05 [TRACE] Session ID: d16d991285d3d791
2017/09/15 15:47:05 [TRACE] CSRF Token: 8JK4531cG-Bdl0vIdkc8Gmb7XmU6MTUwNTQ4MzIyNTExNjkwNDgwMA==
2017/09/15 15:47:05 [TRACE] Session ID: e08b328463658e31
2017/09/15 15:47:05 [TRACE] CSRF Token: U5lwW2zQtfP98SVAyaLW2fAGRbI6MTUwNTQ4MzIyNTIzOTEwNTU4MQ==
2017/09/15 15:47:05 [TRACE] Session ID: dcbbd3d7637d4c1c
2017/09/15 15:47:05 [TRACE] CSRF Token: 1h21smWUoBVKcqeO4k4akpfo_KA6MTUwNTQ4MzIyNTM3NzcxNDgzNA==
2017/09/15 15:47:05 [TRACE] Session ID: c1fb15ed49ed6396
2017/09/15 15:47:05 [TRACE] CSRF Token: NAaIicXoB8BHjndvmHY8QfmGFY06MTUwNTQ4MzIyNTQ2OTIwMjY5Nw==
2017/09/15 15:47:05 [TRACE] Dialing LDAP with security protocol '0' without verifying: false
2017/09/15 15:47:05 [TRACE] LDAP will use BindDN
2017/09/15 15:47:05 [TRACE] Search for LDAP user: xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound as BindDN: cn=xxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxx,dc=xxxxxxx,dc=xxxxxx
2017/09/15 15:47:05 [TRACE] Searching for DN using filter '(cn=xxxxxxxxxx)' and base 'dc=xxxxx'
2017/09/15 15:47:05 [TRACE] Fetching attributes '', '', '', 'mail' with filter '(cn=xxxxxxxx' and base 'cn=xxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx'
2017/09/15 15:47:05 [TRACE] Session ID: 781691565d356bdf
2017/09/15 15:47:05 [TRACE] CSRF Token: voLjdg-i6MisTHswyc4k_KW9Ek46MTUwNTQ4MzIyNTQ4MzczNzM4OQ==
2017/09/15 15:47:05 [TRACE] Session ID: 263128dba04e1c20
2017/09/15 15:47:05 [TRACE] CSRF Token: uCnyBjbiW6_eOk8FM8zXqn73m6k6MTUwNTQ4MzIyNTQ4MzgzNDMyMA==
2017/09/15 15:47:05 [TRACE] Binding with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound successfully with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Session ID: 2ea756151f43bfe4
2017/09/15 15:47:05 [TRACE] CSRF Token: mJhb2ME3iVAXxnHQMw19TbAgQ446MTUwNTQ4MzIyNTQ3Mjk1OTE2MA==
2017/09/15 15:47:05 [TRACE] Dialing LDAP with security protocol '0' without verifying: false
2017/09/15 15:47:05 [TRACE] LDAP will use BindDN
2017/09/15 15:47:05 [TRACE] Search for LDAP user: xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound as BindDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Searching for DN using filter '(cn=xxxxxxxx)' and base 'dc=xxxxxxxx'
2017/09/15 15:47:05 [TRACE] Fetching attributes '', '', '', 'mail' with filter '(cn=xxxxxxxx)' and base 'cn=xxxxxxxxx,ou=xxxxxxxxx,ou=xxxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxxx,dc=xxxxxxxx'
2017/09/15 15:47:05 [TRACE] Binding with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound successfully with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxxx,dc=xxxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] HTTPGit - Authenticated user: xxxxxxxx
2017/09/15 15:47:05 [TRACE] Session ID: 20f1d0f2ea2df3f9
2017/09/15 15:47:05 [TRACE] CSRF Token: xRuFybH5i4BwKyKtLh0_7xgcMhM6MTUwNTQ4MzIyNTYxODY3NjU4MA==
2017/09/15 15:47:05 [TRACE] Dialing LDAP with security protocol '0' without verifying: false
2017/09/15 15:47:05 [TRACE] LDAP will use BindDN
2017/09/15 15:47:05 [TRACE] Search for LDAP user: xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound as BindDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Searching for DN using filter '(cn=xxxxxxxx)' and base 'dc=xxxxxxxx'
2017/09/15 15:47:05 [TRACE] Fetching attributes '', '', '', 'mail' with filter '(cn=xxxxxxxxx)' and base 'cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx'
2017/09/15 15:47:05 [TRACE] Binding with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] Bound successfully with userDN: cn=xxxxxxxx,ou=xxxxxxxx,ou=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx,dc=xxxxxxxx
2017/09/15 15:47:05 [TRACE] HTTPGit - Authenticated user: xxxxxxxx
2017/09/15 15:47:05 [TRACE] Session ID: 20f1d0f2ea2df3f9
2017/09/15 15:47:05 [TRACE] CSRF Token: xRuFybH5i4BwKyKtLh0_7xgcMhM6MTUwNTQ4MzIyNTYxODY3NjU4MA==
2017/09/15 15:47:05 [TRACE] TriggerTask 'sapbti_artmas/master' by 'xxxxxxxx'
2017/09/15 15:47:05 [TRACE] DeliverHooks [repo_id: 107]
2017/09/15 15:47:05 [TRACE] AddTestPullRequestTask [head_repo_id: 107, head_branch: master]: finding pull requests
2017/09/15 15:47:05 [TRACE] AddTestPullRequestTask [base_repo_id: 107, base_branch: master]: finding pull requests
`

Attached is the corresponding part of the post-receive Logfile
[post-receive.txt](https://github.com/gogits/gogs/files/1306667/post-receive.txt)

**Comment by Corben78** [comment]
I'm having the same issue. We were using gogs 0.9.141 before, here everything worked fine. After upgrading to 0.11.29 we had first issues with pushing in general, as the pre-receive hook failed. They were automatically created during the first start of the new version after the upgrade. I was able to solve it by deleting all created hooks, as mentioned here: https://github.com/gogits/gogs/issues/4211#issuecomment-283185901

I kept only two custom post-receive hooks, which I manually re-created through the gogs web-interface.
They were just now tested and give the same error.

So when issuing `git push` on the cli I get again:
```bash
remote: Gogs: Internal error
```
I have two instances of gogs 0.11.29 running, both are using reverse proxy. First is without a sub-url and gives in the gogs error.log:
`[Macaron] 2017-09-18 12:50:32: Completed POST /user/repo.git/git-receive-pack 401 Unauthorized in 5.43151ms`
With the sub-url I don't get this error, but both show in the `log/hooks`-subfolder:
```
2017/09/18 13:48:21 [FATAL] [...its/gogs/cmd/hook.go:259 runHookPostReceive()] Fail to execute custom post-receive hook: fork/exec /path/to/gogs-repositories/user/repo.git/custom_hooks/post-receive: exec format error
```
We're using apache 2.2 with reverse proxy. Version 0.9.141 is working fine with custom git hooks, 0.11.29 is having this issue.

They are no issues with pushing or cloning in general, just when having custom git hooks.
